source/_posts/Unit-Testing-improvements.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Unit Testing improvements
+date: 2020-03-15 09:23:54
+tags:
+  - Unit Testing
+  - Unit Testing improvements
+  - xUnit
+  - Mocking
+  - Unit Test Naming
+  - Clean Code
+---
+
+
+Tests are stories we tell the next generation of programmers on a project. If you are to lazy to write them well consider moving to a different profession. This isn't a job for you.
+
+Source: [Roy Osherove - The Art of Unit Testing](https://www.manning.com/books/the-art-of-unit-testing-second-edition)
+
+In a world where we use CI / CD pipelines to automate our releases unit tests are an essential part of a successful software project. Not only do they help us to debug the code we are writing faster but they also help us to validate the code before releasing it to our users.
+
+In this post, we will go over the steps you can take to improve your unit tests and therefore create a better product in the end.
+
+# Naming
+
+Using a good naming convention for a unit test does not necessarily help you to write a better product but it will help your team to identify clearly what unit is tested and what state the unit under test is in.
+
+One of the most know conventions that are currently used is `[MethodUnderTest]_[Scenario]_[ExpectedResult]`. While its not a bad naming convention it is limit. Writing the naming of the unit test this way leaves out some imported details and might not make it clear to other programmers what the test is trying to test. With naming it's better to tell the story of the test. Let's look at two examples.
```

Comment:
There are 3 ways of saying **It is ** that are being used: It is, it's and its. Maybe pick one for consistency :)

Reply:
Keep in mind that there is a difference between it's and its